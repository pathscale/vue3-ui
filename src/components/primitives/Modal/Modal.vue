<script>
import { ref, onMounted, watchEffect } from 'vue'

export default {
  name: 'VModal',
  props: {
    modelValue: {
      type: Boolean,
      required: true,
    },
    card: Boolean
  },
  emits: ['update:modelValue'],
  setup(props, { emit }) {
    const active = ref(props.modelValue)

    onMounted(() => {
      emit('update:modelValue', active)
    })

    watchEffect(() => {
      active.value = props.modelValue
    })

    const close = () => {
      emit('update:modelValue', false)
    }
    return { active, close }
  },
}
</script>

<template>
  <div class="modal" :class="{ 'is-active': active }">
    <div class="modal-background" @click="close" />
    <template v-if="!card">
      <div class="modal-content">
        <slot />
      </div>
      <button class="modal-close" aria-label="close" @click="close" />
    </template>
    <div v-else class="modal-card">
      <header class="modal-card-head">
        <p class="modal-card-title">
          <slot name="title" />
        </p>
        <button class="delete" aria-label="close" @click="close" />
      </header>
      <div class="modal-card-body">
        <slot name="content" />
      </div>
      <footer class="modal-card-foot">
        <slot name="footer" />
      </footer>
    </div>
  </div>
</template>
