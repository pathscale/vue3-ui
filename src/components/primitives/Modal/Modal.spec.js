import { shallowMount } from '@vue/test-utils'
import Modal from './Modal.vue'

let wrapper

describe('Modal', () => {
  beforeEach(() => {
    wrapper = shallowMount(Modal)
  })

  it('is called', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
