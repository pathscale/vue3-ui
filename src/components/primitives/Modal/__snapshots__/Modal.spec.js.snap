// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render correctly 1`] = `
<div class="modal">
  <div class="modal-background"></div>
  <div class="modal-content"></div><button class="modal-close" aria-label="close"></button>
</div>
`;
